course-mw/enrolment-actor/src/main/scala/org/sunbird/aggregate/CollectionSummaryAggregate.scala

```diff
@@ -0,0 +1,177 @@
+import javax.ws.rs.core.MediaType
+import org.apache.commons.lang3.StringUtils
+import org.apache.http.HttpHeaders
+import org.joda.time.format.DateTimeFormat
+import org.joda.time.{DateTime, DateTimeZone}
+import org.sunbird.actor.base.BaseActor
+import org.sunbird.cache.util.RedisCacheUtil
+import org.sunbird.common.models.response.Response
+import org.sunbird.common.models.util.{JsonKey, ProjectLogger, ProjectUtil, TelemetryEnvKey}
+import org.sunbird.common.request.Request
+import org.sunbird.learner.util.Util
+
+import scala.collection.JavaConverters.iterableAsScalaIterableConverter
+
+
+class CollectionSummaryAggregate @Inject()(implicit val cacheUtil: RedisCacheUtil) extends BaseActor {
+  val ttl: Int = if (StringUtils.isNotBlank(ProjectUtil.getConfigValue("collection_summary_agg_cache_ttl"))) ProjectUtil.getConfigValue("collection_summary_agg_cache_ttl").toInt else 60
+  val dataSource: String = if (StringUtils.isNotBlank(ProjectUtil.getConfigValue("collection_summary_agg_data_source"))) ProjectUtil.getConfigValue("collection_summary_agg_data_source") else "telemetry-events-syncts"
+  val stateLookUpQuery = "{\"type\":\"extraction\",\"dimension\":\"derived_loc_state\",\"outputName\":\"state\",\"extractionFn\":{\"type\":\"registeredLookup\",\"lookup\":\"stateLookup\",\"replaceMissingValueWith\":\"Unknown\"}}"
+  val districtLookUpQuery = "{\"type\":\"extraction\",\"dimension\":\"derived_loc_district\",\"outputName\":\"district\",\"extractionFn\":{\"type\":\"registeredLookup\",\"lookup\":\"districtLookup\",\"replaceMissingValueWith\":\"Unknown\"}}"
+  val response = new Response()
+  val gson = new Gson
+
+  override def onReceive(request: Request): Unit = {
+    Util.initializeContext(request, TelemetryEnvKey.BATCH)
+    val filters = request.getRequest.get(JsonKey.FILTERS).asInstanceOf[util.Map[String, AnyRef]]
+    val groupByKeys = request.getRequest.getOrDefault(JsonKey.GROUPBY, new util.ArrayList[String]()).asInstanceOf[util.ArrayList[String]].asScala.toList
+    val batchId = filters.get(JsonKey.BATCH_ID).asInstanceOf[String]
+    val collectionId = filters.get(JsonKey.COLLECTION_ID).asInstanceOf[String]
+    val dateTimeFormate = DateTimeFormat.forPattern("yyyy-MM-dd")
+    val presentDate = dateTimeFormate.print(DateTime.now(DateTimeZone.UTC))
+    val fromDate = dateTimeFormate.print(DateTime.now(DateTimeZone.UTC).minusDays(7))
+    val defaultDate = s"$fromDate/$presentDate"
+    val key = getCacheKey(batchId = batchId, request.getRequest.getOrDefault("intervals", defaultDate).asInstanceOf[String], groupByKeys)
+    try {
+      val redisData = cacheUtil.get(key)
+      val result: String = Option(redisData).map(value => if (value.isEmpty) {
+        getResponseFromDruid(batchId = batchId, courseId = collectionId, date = defaultDate, groupByKeys = groupByKeys)
+      } else {
+        value
+      }).getOrElse(getResponseFromDruid(batchId = batchId, courseId = collectionId, date = defaultDate, groupByKeys = groupByKeys))
```

Comment:
getOrElse is not required here. Directly call get.

Reply:
The reason I have added "getOrElse" to avoid the null pointer and emtpy string issue. if the redis returns empty string or null.